### ISSUE TITLE
@Autowire WebDriver lifecycle not correct

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-boot#6641]
In Spring Boot 1.4.0.RELEASE the testing support does not invoke `WebDriver.quit()` between tests. It should also create a new instance for every test method.

The `@Autowire` `WebDriver` should add a hook so that after every test `WebDriver.quit()` is invoked and a new instance is created at the beginning of each tests.

**Comment by tomwhoiscontrary** [comment]
:+1: Thanks!

**Comment by alexhutnik** [comment]
@philwebb @rwinch:

How might I take advantage of this lifecycle if I'm wiring in my own WebDriver?

I have a bean definition like so:
```
@Bean
WebDriver getDriver(Environment environment){
    switch (environment.getProperty("driver")){
        case "chrome":
            return new ChromeDriver();
        case "htmldriver":
            return new HtmlUnitDriver();
        case "phantomjs":
            return new PhantomJSDriver();
        default:
            return new HtmlUnitDriver();
    }
}
```

This allows me to specify from Maven (via profiles) or from my IDE (via args appended in JUnit runner) which WebDriver I want to use.

My web integration tests extend a class like this:
```
@RunWith(SpringRunner.class)
@SpringBootTest(classes = ApplicationLauncher.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public abstract class AbstractWebIntegrationTest {
    @Autowired
    protected WebDriver driver;
}
```

And a sample test:
```
public class EnrollmentIntegrationTest extends AbstractWebIntegrationTest {

    @Autowired
    EnrollmentPage enrollmentPage;

    @Test
    public void should_process_valid_enrollment(){
        enrollmentPage.open(url("/enroll"));
        enrollmentPage.createEnrollment(random(EnrollmentForm.class));
    }

    @Test
    public void should_process_valid_enrollment2(){
        enrollmentPage.open(url("/enroll"));
        enrollmentPage.createEnrollment(random(EnrollmentForm.class));
    }

    @Test
    public void should_process_valid_enrollment3(){
        enrollmentPage.open(url("/enroll"));
        enrollmentPage.createEnrollment(random(EnrollmentForm.class));
    }
}
```

I'm wondering how the driver can get reset at the end of a test method invocation.  I've tried this and it seems like what is happening is `WebDriverContextCustomizerFactory` is detecting my WebDriver and setting the `webDriver` scope on it.  Then after the first test is run `WebDriverTestExecutionListener` is resetting the scope (which calls `quit()` on the driver).  But when the next method is invoked, the `@Autowired` `WebDriver` is not re-initialized using the bean definition above.

**Comment by philwebb** [comment]
@alexhutnik 

How is the `EnrollmentPage` defined? You could either try adding `@Scope("webDriver")` to it's `@Bean` method so that it gets cleared out when the scope is reset or perhaps change it's internals to use `@Autowired javax.inject.Provider<WebDriver> driver` then call `driver.get` each time you need the driver.

**Comment by nikit-cpp** [comment]
Hi, I have same "approach" as @alexhutnik: I wrired my own WebDriver bean with my own lifecycle: I want to start and stop WebDriver once, at Junit tests startup and stop after all tests done.
All works Ok, but in Spring Boot 1.4.1 appear WebDriverTestExecutionListener which break my wanted behaviour.
I tried to exclude it with
```spring.autoconfigure.exclude: org.springframework.boot.test.autoconfigure.web.servlet.WebDriverTestExecutionListener```
but it not excluded.

How I can exclude package-local class WebDriverTestExecutionListener ?
